### PR TITLE
[lldb] Unify Platform::ResolveExecutable

### PR DESCRIPTION
**Pull request by JDevlieghere**
The Platform class currently has two functions to resolve an executable: `ResolveExecutable` and `ResolveRemoteExecutable`. The former strictly deals with local files while the latter can handle potentially remote files. I couldn't figure out why the distinction matters, at the latter is a super-set of the former.

To make things even more confusion, we had a similar but not identical implementation in RemoteAwarePlatform where its implementation of `ResolveExecutable` could handle remote files. To top it all off, we had copy-pasted implementation, dead code included in `PlatformAppleSimulator` and `PlatformRemoteDarwinDevice`.

I went ahead and unified all the different implementation on the original `ResolveRemoteExecutable` implementation. As far as I can tell, it should work for every other platform, and the test suite (on macOS) seems to agree with me, except for a small wording change.

### CODE REVIEW

**Review by jasonmolenda — APPROVED** [review]
LGTM, nice cleanup.

File: lldb/include/lldb/Target/Platform.h [hunk 1]
```diff
@@ -1004,10 +1004,7 @@ class Platform : public PluginInterface {
 
   virtual const char *GetCacheHostname();
 
-  virtual Status
-  ResolveRemoteExecutable(const ModuleSpec &module_spec,
-                          lldb::ModuleSP &exe_module_sp,
-                          const FileSpecList *module_search_paths_ptr);
+  /// The method is virtual for mocking in the unit tests.
```
Comment (labath) [inline lldb/include/lldb/Target/Platform.h:1007]: Which function is this comment referring to (if I'm reading the diff right, it's just floating there)?

**Review by labath — APPROVED** [review]
Good stuff.

File: lldb/include/lldb/Target/Platform.h (hunk 1 above)
Comment (JDevlieghere) [reply lldb/include/lldb/Target/Platform.h:1007]: This comment is outdated. It was for `ResolveRemoteExecutable` which was virtual but only overridden in the unit test. I'll remove it.

**Comment by slydiman** [comment]
It seems this patch breaks the following tests on cross setups: 
```
Failed Tests (6):
  lldb-api :: commands/process/attach/TestProcessAttach.py
  lldb-api :: commands/process/detach-resumes/TestDetachResumes.py
  lldb-api :: functionalities/dyld-exec-linux/TestDyldExecLinux.py
  lldb-api :: functionalities/dyld-launch-linux/TestDyldLaunchLinux.py
  lldb-api :: functionalities/exec/TestExec.py
  lldb-api :: functionalities/thread/create_after_attach/TestCreateAfterAttach.py
```
Note the behavior is the same for Windows and Linux hosts with Linux AArch64 target.
Probably removing `RemoteAwarePlatform::ResolveExecutable` causes the issue.

For example here is the log for lldb-api :: commands/process/attach/TestProcessAttach.py
```
Command Output (stdout):
--
Setting up remote platform 'remote-linux'
Connecting to remote platform 'remote-linux' at 'connect://test1:1234'...
Connected.
Setting remote platform working directory to '/home/ubuntu/lldb-tests'...
Skipping the following test categories: ['dsym', 'gmodules', 'debugserver', 'objc', 'lldb-dap']
--
Command Output (stderr):
--
PASS: LLDB (C:\\lldb-test-scripts\build-lldb\bin\clang.exe-aarch64) :: test_attach_to_process_by_id (TestProcessAttach.ProcessAttachTestCase.test_attach_to_process_by_id)
PASS: LLDB (C:\\lldb-test-scripts\build-lldb\bin\clang.exe-aarch64) :: test_attach_to_process_by_id_autocontinue (TestProcessAttach.ProcessAttachTestCase.test_attach_to_process_by_id_autocontinue)
FAIL: LLDB (C:\\lldb-test-scripts\build-lldb\bin\clang.exe-aarch64) :: test_attach_to_process_by_id_correct_executable_offset (TestProcessAttach.ProcessAttachTestCase.test_attach_to_process_by_id_correct_executable_offset)
PASS: LLDB (C:\lldb-test-scripts\build-lldb\bin\clang.exe-aarch64) :: test_attach_to_process_by_name (TestProcessAttach.ProcessAttachTestCase.test_attach_to_process_by_name)
PASS: LLDB (C:\lldb-test-scripts\build-lldb\bin\clang.exe-aarch64) :: test_attach_to_process_from_different_dir_by_id (TestProcessAttach.ProcessAttachTestCase.test_attach_to_process_from_different_dir_by_id)
======================================================================
FAIL: test_attach_to_process_by_id_correct_executable_offset (TestProcessAttach.ProcessAttachTestCase.test_attach_to_process_by_id_correct_executable_offset)
   Test that after attaching to a process the executable offset
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\lldb-test-scripts\lldb\test\API\commands\process\attach\TestProcessAttach.py", line 119, in test_attach_to_process_by_id_correct_executable_offset
    lldbutil.run_break_set_by_file_and_line(
  File "C:\\lldb-test-scripts\lldb\packages\Python\lldbsuite\test\lldbutil.py", line 378, in run_break_set_by_file_and_line
    check_breakpoint_result(
  File "C:\\lldb-test-scripts\lldb\packages\Python\lldbsuite\test\lldbutil.py", line 592, in check_breakpoint_result
    test.assertTrue(
AssertionError: False is not true : Expecting 1 locations, got 0.
Ran 5 tests in 71.809s
FAILED (failures=1)
```

**Comment by clayborg** [comment]
We think this broke Android debugging on our end. We are investigating.

**Comment by clayborg** [comment]
This did indeed break Android debugging. We reverted this in our repo and functionality was restored

**Pull request merged**